docs/guides/hmac-authentication-in-ruby.md

```diff
@@ -0,0 +1,175 @@
+If you don't have the file, make one with the following content: 
+
+```
+source "https://rubygems.org"
+
+git_source(:github) { |repo_name| "https://github.com/#{repo_name}" }
+```
+
+We will need to add faraday to Gemfile [Faraday](https://lostisland.github.io/faraday) for  HTTP requests to the Mekari API.
+```
+# Gemfile
+gem 'faraday'
+```
+
+And we will put all credentials on .env since  we don't want to Mekari API client ID and client secret exposed directly on the code. To load the .env, we will need [Doetenv](https://github.com/bkeepers/dotenv) .
+
+```
+# Gemfile
+gem 'dotenv'
+```
+
+## Making API Request
+{: .fw-300 }
+
+Using Faraday and Dotenv that we have installed earlier, we are going to setup Ruby script that will perform API request to one of Klikpajak API endpoint which is [create sales invoice](https://documenter.getpostman.com/view/17365057/U16hrR5d#63ba32aa-0f91-44a5-ab40-a0da6a8bf608) (`https://api.mekari.com/v2/klikpajak/v1/efaktur/out`). 
+
+We create the script called `main.rb` then use Faraday to perform the HTTP POST request. This is how the script will looks like:
+
+```
+path = '/v2/klikpajak/v1/efaktur/out/'
+headers = { 'X-Idempotency-Key' => '1234' }
+response = Faraday.post("#{ENV['MEKARI_API_BASE_URL']}/#{path}", nil, headers)
+
+puts "Got response with status: #{response.status}, body: #{response.body}"
+```
+
+We can run the script on the console by typing `dotenv -f .env ruby main.rb`. Because we just created a plain HTTP POST request, it will has an error response similar to this: 
+
+```
+$ dotenv -f .env ruby main.rb
+Got response with status: 401, body: {"message":"Unauthorized"}
+```
+
+## Creating HMAC Signature
+{: .fw-300 }
+
+[The signature](/docs/kb/authentication/hmac#generating-signature) is one of the requirements for forming an API request with HMAC Authentication. The signature is an HMAC256 representation of the request line (a combination of the request method, the request path, and `HTTP/1`.1) and the `Date` header in [RFC 7231](https://www.ietf.org/rfc/rfc7231.txt) format. Carbon will be used to generate the date string for us. The signature must then be converted into a Base64 string so that it can be attached to the `Authorization` header.
+
+The code will look like this: 
+
+```
+<?php // main.php
+
+use Carbon\Carbon;
+
+// ... the rest of the code
+
+datetime = Time.now.httpdate
+request_line = "POST /v2/klikpajak/v1/efaktur/out HTTP/1.1"
+payload = [datetime, request_line].join("\n")
+digest = OpenSSL::HMAC.digest(OpenSSL::Digest.new('sha256'), 'YOUR_MEKARI_API_CLIENT_SECRET', payload)
+signature = Base64.strict_encode64(digest)
+
+puts "hmac username=\"YOUR_MEKARI_API_CLIENT_ID", algorithm=\"hmac-sha256\", headers=\"date request-line\", signature=\"#{signature}\""
+```
+
+If you replace `datetime` with `Wed, 10 Nov 2021 07:24:29 GMT` and run the code, you will get the following result.
+
+```
+$ dotenv -f .env ruby main.rb
```

Comment:
@syafdia Since `.env` file is created in later paragraph, this code should not contain `dotenv -f .env`. It should only `ruby main.rb`.